addons/sale_purchase/tests/test_sale_purchase.py

```diff
@@ -364,3 +364,35 @@ def test_service_to_purchase_multi_company(self):
         # FIXME: same sudo issue as above
         order2.sudo().with_company(company_2).action_confirm()
         self.assertTrue(order2.purchase_order_count)
+
+    def test_service_to_purchase_branch_tax_propagation(self):
+        """
+            Test that taxes are passed from the SO to the PO for service products in branch companies
+        """
+        branch = self.env['res.company'].create({
+            'name': "Branch Company",
+            'parent_id': self.env.company.id,
+        })
+        service_product = self.env['product.product'].with_company(branch).create({
+            'name': "Branch Out-sourced Service",
+            'standard_price': 200.0,
+            'type': 'service',
+            'invoice_policy': 'delivery',
+            'taxes_id': self.company_data['default_tax_sale'],
+            'service_to_purchase': True,
+            'seller_ids': [Command.create({
+                'partner_id': self.partner_b.id,
+                'min_qty': 1,
+                'price': 100,
+            })],
+        })
+        so_form = Form(self.env['sale.order'].with_company(branch))
+        so_form.partner_id = self.partner_a
+        with so_form.order_line.new() as line:
+            line.product_id = service_product
+            line.product_uom_qty = 1
+        so = so_form.save()
```

Comment:
Avoid the use of `Form` when possible
```suggestion
        so = self.env['sale.order'].create({
            'partner_id': self.partner_a.id,
            'order_line': [
                Command.create({
                    'product_id': service_product.id,
                }),
            ],
        })
```

Reply:
Sure thing, just gotta specify the company aswell
> so = self.env['sale.order'].with_company(branch).create({